### DISCUSSION TITLE
Using a misorientation vector in specimen reference frame

### DISCUSSION THREAD

**Discussion by angus0997** [mtex-toolbox/mtex#2169] [Ask Anything] [answered]
Hi Everyone,

I would like to define a misorientation vector, as used in this paper: https://doi.org/10.1107/S1600576719004035.
The misorientation vector should link two orientations in the _specimen_ reference frame, and be used to convert the one orientation into the other. The norm of this vector should be the misorientation angle, and the direction of this vector should be the misorientation axis, in specimen coordinates.

Working in the _crystal_ reference frame, we can do the following:
`ori1=ebsd(1).orientations;`
`ori2=ebsd(2).orientations;`
`mori=inv(ori1)*ori2;`
 And now `mori` defines the misorientation in the crystal reference frame, and can be used to convert ori1 to ori 2: `ori1*mori = ori2;`.

Working in the _specimen_ reference frame, we can do the following:
`mori_axis = axis(ori1,ori2) % Misorientation axis in Specimen frame`
`mori_angle = angle(ori1,ori2)/degree % Corresponding misorientation angle`
Now a misorientation vector can be defined:
`mori_vec = mori_angle * mori_(axis) `

My question is: How can this misorientation vector be used to convert ori1 to ori2?

Best,
Angus

**Comment by kilir** [comment]
Hi,
one way would be the following:

```
rot = rotation.byAxisAngle(-mori_vec,norm(mori_vec));
o2new = rot * ori1
angle(ori2,o2new)/degree
```

Cheers,

Rüdiger

**Reply by angus0997** [reply]
Hi Rüdiger, thank you for your reply. 
I had tried using rotation.byAxizAngle this way, but it seems not to work. I do not know why.
For example, picking the following two arbitrary orientations I get a 25.8° angle between ori2 and o2new:
 ```matlab
ori1 = orientation.byEuler(30*degree,50*degree,10*degree,cs);
ori2 = orientation.byEuler(100*degree,50*degree,60*degree,cs);
mori_axis = axis(ori1,ori2); % Misorientation axis in Specimen frame
mori_angle = angle(ori1,ori2)/degree; % Corresponding misorientation angle
mori_vec = mori_angle * mori_axis;
rot = rotation.byAxisAngle(-mori_vec,norm(mori_vec));
o2new = rot * ori1;
angle(ori2,o2new)/degree
Warning: The rotation angle appears to be quite large. Maybe you forgot '* degree' to switch from degree to radians. 
> In axis2quat (line 36)
In rotation.byAxisAngle (line 17) 

ans =

   25.7880
```
Best,
Angus

**Reply by kilir** [reply, marked as answer]
Hi,
yes, for `mori_angle` you should skip the `degree` and stay in radians or convert back to radians when using the norm.
Cheers,
Rüdiger

**Reply by angus0997** [reply]
Thank you, this works: 
```matlab
ori1 = orientation.byEuler(30*degree,50*degree,10*degree,cs);
ori2 = orientation.byEuler(100*degree,50*degree,60*degree,cs);
mori_axis = axis(ori1,ori2); % Misorientation axis in Specimen frame
mori_angle = angle(ori1,ori2); % Corresponding misorientation angle
mori_vec = mori_angle * mori_axis;
rot = rotation.byAxisAngle(-mori_vec,norm(mori_vec));
o2new = rot * ori1;
angle(ori2,o2new)/degree

ans =

     0
```
